### DISCUSSION TITLE
Reorganizing Settings Menu(s) for improved UX.

### DISCUSSION THREAD

**Discussion by Lupum1001** [NightscoutFoundation/xDrip#1991] [Ideas]
Wanted to start up a discussion about re-organizing some settings menus. At the moment, the main hierarchy is a bit wonky and I find it's hard to find things. I was considering drafting up a formal before/after proposal with screenshots and so forth, and was wondering if other people were interested.

### Proposed Scope of changes:

- In an attempt to limit the complexity/scope this time around, I'd be hoping to _move_ the existing settings around, 
but not add/remove/hide any functionality (i.e. a confusing list of checkboxes would remain a confusing list of checkboxes 😉).
- Any ideas for _improvements_ to UI elements are of course welcome, but can probably be left for a separate pull request.

### Examples of things I'm thinking of:
- I noticed at least 4 different "overflow" designations on the page below: ("other settings", "less common settings", "Xdrip+ Extra Settings", and "experimental section".
- Perhaps it would make sense to consolidate some menus (Glucose Meters, Smart Watches, Insulin Pens, Dexcom, Libre, etc) into a "connected devices" sub-menu instead? It's not without drawbacks though, so there might be another option.

![image](https://user-images.githubusercontent.com/52087143/153538652-7fe7b8df-1f25-4407-8277-dcdfc55a918f.png)

**Comment by Navid200** [comment]
It may be easier to make small changes in steps rather than making too many changes at once.  

It is easier to explain a small change and it is easier for the person reviewing it to understand the proposal.
If you break down your proposal into many small changes, at the end you may get them all approved.  But, even if you only get some of them approved, it is still better than getting none.

If you have an array of changes, in one proposal, and the person reviewing it disagrees with only one, that is enough not to approve the entire proposal even if he/she agrees with another one of the included items.  In other words, by combining the items, you are imposing the weakness of the weakest item on all the other items.

Of course, on the other end of the spectrum, if you can come up with a perfect single proposal that gets approved, including everything you want to do, you may save some time.  But, you will be taking a chance.
Just look at the list of open PRs.  The onus is on the person making the proposal to convince the reviewers.  It's not the other way around.  
You can see examples of complicated proposals that have not been approved, for years, and the requested changes are too difficult to implement.

**Comment by ChanceHarrison** [comment]
https://github.com/NightscoutFoundation/xDrip/issues/1904 seems to be related to this matter.

**Comment by Lupum1001** [comment]
So starting with something small to test the waters:

### **Part 1** - Consolidate software update and backup/restore menus
_UPDATE:  I've submitted the following pull request to implement these changes:  https://github.com/NightscoutFoundation/xDrip/pull/2057_

In the settings menu create a new submenu. I've re-purposed a previous icon, but I suppose we can always come up with a new one.

![image](https://user-images.githubusercontent.com/52087143/154835725-96338a43-fcfe-49c5-a3ab-a355f5096af5.png)
The following menus can be moved into that submenu, as desired:

- "End User License Agreement" - (unless there's some legal reason to keep it up there after the initial install/acceptance?)
- "XDrip+ Update Settings"
- "Show Settings QR Codes"
- "Auto Configure" (barcode scanner) - _*** I wasn't sure about this one, since they're nowhere near each other, but I didn't see any other complement to the QR generator. Is this page intended for XDrip QR codes, or is it for some other cloud/data sync feature?_
-  "Load / Save settings to SD card"


Feedback/Suggestions are appreciated. This won't sufficiently declutter the settings page on its own, but it's a decent bite-sized chunk to start off with.

**P.S.** I noticed that the EULA is also tucked under "predictive simulation settings".

**Comment by Lupum1001** [comment]
After some investigating, there appear to be two barcode scanners. The other one only shows up when you select "G4 Share Receiver" as your hardware source. The preference key for that hidden one is `scan_share2_barcode`. The main scanner with the icon has the preference key `auto_configure`.
Based on the original repository commit, the scanner was intended to be used with these nightscout code generators:
https://nightscout.github.io/pages/configure/

**Reply by tolot27** [reply]
Just for the records: https://github.com/NightscoutFoundation/xDrip/pull/1151#issuecomment-892123468

**Comment by jbr7rr** [comment]
I was also thinking about cleaning up the settings menu,

One thing that bothers me a bit is that the sensor specific settings are scattered all over the settings menu. E.g. Libre2 and dexcom checkboxes can be found in the same menu.

I would like to propose to consolidate all the settings into a specific settings menu for the active settings, much like the G5/G6 debug settings menu.

For the Libre it would look something like this

MAIN MENU:
- Hardware source
   X Libre selected
- Libre settings
    - NFC scan functions
        - Same as current menu..
    - Display settings
        - Smooth sensor noise
        - Show Libre trend
        - Show libre trend in graph
        - Show raw values in graph
        - show sensor info in status


I understand that this is outside your scope, but if this sounds like a good idea I can create a separate discussion and/or try something in code and create a PR.

**Reply by Lupum1001** [reply]
I don't have any experience with the Libre, but the device-specific stuff seems reasonable. I'm not _quite_ sure about more generic things like "smooth sensor noise" though.
